README.md

```diff
@@ -53,40 +96,78 @@ which can obscure illegal activities,
 such as unauthorized fishing activity or
 unauthorized transshipments [[1]](#1)[[2]](#2).
 
+</div>
+
 ## Definition of gap
 
-We define an **AIS** gap event when the period of time between
+<div align="justify">
+
+We create an **AIS** **gap** event when the period of time between
 consecutive known good **AIS** positions from a single vessel,
-after de-noise and de-spoofing),
-exceeds a configured `threshold` (typically 12 hours).
+after de-noise and de-spoofing,
+exceeds a configured `threshold` (typically 6 hours).
+The `start/end` position messages of the gap are called `OFF/ON` messages,
+respectively.
 
 When the period of time between **last** known good position
-and the present time exceeds the `threshold`,
-we call it an **open** gap event.
+and the last time of the current day exceeds the `threshold`,
+we create an **open gap** event.
+In that case, the gap will not have an `ON` message,
+until it is **closed** in the future when new data arrives.
+
+Additionally,
+the gaps that surpass the `threshold`
+but satisfies the condition `distance_from_shore_m = 0`
+in the `OFF` or the `ON` message are filtered out from the results.
+
+</div>
+
+## Some results
+
+These are some results for 2021-2023.
+
+<div align="center">
+
+![alt text](analysis/gaps.svg)
+
+</div>
 
 ## Usage
 
 ### Installation
 
 We still don't have a package in PYPI.
-First, clone the repository.
 
-Then
+First, clone the repository.
+Then run inside a virtual environment
+```shell
+make install
+```
+Or, if you are going to use the dockerized process, build the docker image:
 ```shell
-pip install .
+make build
 ```
 
-If you want to use apache beam integration:
+In order to be able to connect to BigQuery, authenticate and configure the project:
 ```shell
-pip install .[beam]
+docker compose run gcloud auth application-default login
+docker compose run gcloud config set project world-fishing-827
+docker compose run gcloud auth application-default set-quota-project world-fishing-827
 ```
 
-### Using from python code
+### Gap detection low level process
+
+<div align="justify">
+
+The gap detection core process takes as input a list of **AIS** messages.
+Mandatory fields are `["timestamp", "distance_from_shore_m"]`.
+At this level, messages with different **SSVIDs** are not distinguished.
```

Comment:
At first I was confused by this because I didn't realise that this is a distinction from the high level process. Might make sense to add a short section above that explains that there's also a high level process (I know, it's all explained further down but I think the grouping by ssvid is essential, even though we might change that in the future).

Reply:
I think I will remove that sentence since it can be confusing. 
Maybe I will reorder sections afterwords.